include/spdlog/details/line_logger_impl.h

```diff
@@ -46,6 +46,11 @@ inline spdlog::details::line_logger::~line_logger()
 #endif
         _callback_logger->_log_msg(_log_msg);
     }
+
+    if (_log_msg.level >= _callback_logger->_flush_level)
+    {
+        _callback_logger->flush();
+    }
```

Comment:
it better be inside the _log_msg() function. otherwise it get called always even if the log is filtered in this level